### ISSUE TITLE
Make addons project github labels useful

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons#305]
Filing based on feedback in irc from @EnTeQuAk  + @diox

- How are we using labels across the projects currently? 
- How can we consolidate/sync labels them across projects?
- Can we add more detail to the labels so they're more descriptive 
  - Should that be on a wiki page or can we use some clever prefixes.
- Can we provide better guidelines for colours so they are consistent
  - Possibly something that will be solved by sync.

## TODOS

* [x] Update mozilla/addons
* [x] Update mozilla/addons-server
* [x] Update mozilla/addons-linter
* [x] Update mozilla/addons-frontend
* [x] Update mozilla/amo-validator
* [x] Update mozilla/dispensary
* [x] Update mozilla/web-ext?
* [x] Check waffle is using the right labels
* [x] Update the in progress / pull request ready labels last.
* ~Check addons-bot is using the correct triage label~ filed separately here: https://github.com/mozilla/addons-bots/issues/15

Feel free to add suggestions and comments below.

**Comment by EnTeQuAk** [comment]
Warning: This is long and should probably be a wiki page 😨 

Alrighty… I went through all our projects (addons-server, addons-frontend, amo-validator, addons-linter) and tagged / unified labels.

I didn't touch `mozilla/addons` yet since I wasn't sure we should call it a project or if it needs a more custom label. It should contain the unified list eventually since it's the main point of contact for users though... not so sure.

# Suggested Unified Label List (WIP)
## Types

* Type: User Feedback
* Type: Good First Bug
* Type: Maybe Good First Bug
* Type: Bug
* Type: Feature
* Type: Discussion
* Type: Deprecated Rule (addons-linter / needed?)
* Type: Rule (addons-linter / amo-validator)
* Type: Compat Rule (e.g deprecations etc - linter / validator)

## Impact

* Impact: Code Quality
* Impact: Development Environment / Docker
* Impact: Performance / Operations
* Impact: Regression (or Type?)
* Impact: Firefox Depenceny (both ways, blocks Firefox / is blocked by Firefox?)
* Impact: Automation (existed in addons-frontend, not sure what it was supposed to mean)
* Impact: Security


## Components

* Component: Accessibility
* Component: Ratings (do we need that?)
* Component: Admin
* Component: Addons
* Component: API
* Component: Statistics
* Component: API
* Component: DevHub
* Component: Disco Pane
* Component: Documentation
* Component: Firefox Accounts
* Component: Reviews (isn't that part of DevHub or Addons?)
* Component: i18n
* Component: l10n
* Component: addons-linter
* Component: Reviewer Tools
* Component: Signing
* Component: UI Tests
* Component: Server (addons-frontend)
* Component: Tests
* Component: Tracking
* Component: CLI (addons-linter)
* Component: CSS (addons-linter)
* Component: HTML (addons-linter)
* Component: JavaScript (addons-linter)
* Component: Layout (addons-linter)
* Component: Metadata (addons-linter)
* Component: Schema (addons-linter)


## Platforms (more AMO, Linter, Validator)

* Platform: WebExtension
* Platform: Legacy
* Platform: Android


## Status

* Status: Blocked By Upstream
* Status: Duplicate
* Status: In Progress
* Status: In Testing (is that used?)
* Status: Invalid
* Status: Wontfix
* Status: Needs Info
* Status: Needs More QA
* Status: Needs More Triage
* Status: Needs Ops
* Status: No QA
* Status: Needs UX
* Status: Needs Editorial (I'm guessing for text questions?)
* Status: Not Ready For Merge (used for PRs only)
* Status: Pull Request Ready ("pull request ready" used by waffle, not sure we can change that)
* Status: Triaged
* Status: Verified Fixed
* Status: Blocked / On Hold
* Status: Needs API (addons-frontend)
* Status: Needs Discussion / Feedback


## Priorities

* Priority: Blocks Release
* Priority: Enhancement (Intended to refer to very minor changes that are low priority)
* Priority: P1 (high priority but doesn't block release)
* Priority: P2


## Goals / Project

 * Project: DevHub Landing Page 16Q4
 * Project: Legacy Switch Off 17Q1
 * Project: e10s
 * Project: Redesign (is that a trash-label for everything potentially redesign related?)
 * Project: Style Refresh
 * Project: Version Channel
 * Project: Mobile Pages V1
 * Project: Mobile Pages V2

## Others

* greenkeeper


## Notes

* I'd merge "closed by design" and just use "wontfix" instead since imho the intend is similar.
* MVP label is less used, was a project, should be removed
* "needs rule doc" from addons-linter... not sure what to do about that

# Marked Labels

* `~` - means shared between projects / unified
* `+` - means that it was mostly unique (or initially found here) added to unified list
* `-` - means that it should / could easily be removed
* no labeling / `...` means that I wasn't sure what to do about it

## addons-server

```
+ accessibility
+ add-on ratings
+ admin tools
+ Android
+ API
+ blocked by upstream
+ blocks release
- closed by design
+ code quality
+ development environment
+ devhub landing page v1
+ devhub
+ disco pane
- docker
+ documentation
+ duplicate
+ e10s
+ enhancement
+ feedback
+ Firefox Accounts
+ good first bug
+ greenkeeper
+ i18n
+ in progress
+ in testing
+ invalid
+ Legacy switch off
+ linter
+ maybe good first bug
- MVP
+ needs info
+ needs more qa
+ needs more triage
+ needs no qa
+ needs ux
+ not ready for merge
~ operations
~ perf
+ priority one
+ priority two
+ pull request ready
+ ratings and reviews
+ redesign
+ regression
+ reviewer tools
+ signing
+ style refresh
+ triaged
+ ui tests
- user-generated content (only one issue, or maybe have that as 'Impact'?)
+ verified fixed
+ version channel
+ WebExtensions
+ wont fix
```

## addons-frontend

```
~ accessibility
~ add-on ratings
~ admin tools
~ amo-statistics
- amo (or should that be a component?)
- backlog (that's `triaged`)
+ blocked
~ blocks release
+ bug
~ code quality
~ disco pane
~ documentation
~ duplicate
~ enhancement
- FF50+ (hmm... Project?)
+ firefox-dep
- footgun (doh!)
~ good first bug
+ greenkeeper
... help wanted (how is this being used? Good First Bug?)
~ i18n
~ in progress
~ invalid
~ maybe good first bug
+ mobile pages v1
+ mobile pages v2
- MVP
+ needs api
+ needs automation
+ needs discussion
+ needs editorial
~ needs info
~ needs more qa
~ needs more triage
~ needs no qa
+ needs ops
~ needs ux
~ ops
~ perf
~ priority one
~ priority two
~ pull request ready
- Q4 (should be more detailed and -> Project: ...)
- question (Type: User Feedback / Discussion?)
~ regression
+ security
+ server
+ testing (Component: Tests)
+ tracking
~ triaged
~ uitests
~ upstream-dep
~ UX (Status: Needs UX?)
~ verified fixed
windows (Platform maybe?)
~ wontfix 
```

## amo-validator

```
~ needs-info
~ needs more triage
~ pull request ready
~ qa- (Status: Needs No QA)
~ triaged 
```

## addons-linter

```
~ blocked by upstream
~ blocks release
+ CLI
~ code quality
compatibility ... very unsure, Component, Impact?
+ css
+ deprecated-rule
~ docs
~ duplicate
~ enhancement
~ good first bug
~ greenkeeper
... help wanted (how is this being used? Good First Bug?)
+ html
~ in progress
~ in testing
~ invalid
+ javascript
+ l10n
+ layout 
~ maybe good first bug
+ metadata
- MVP
~ needs discussion
~ needs info
~ needs more triage
~ needs no qa
needs rule doc
~ priority one
~ priority two
~ pull request ready
~ question
~ regression
~ rule
+ schema
~ security
~ tests
~ triaged
~ verified fixed
web (huh?)
~ wontfix
- yak-shaving (code quality...)
```

**Comment by EnTeQuAk** [comment]
Also, there is a nice (and probably more to find) project to manage labels: https://github.com/jasonbellamy/git-label-packages - per project basis though but I feel like we will have customizations anyway.

**Comment by EnTeQuAk** [comment]
Also one thing I recently read about and found to be useful to think about: Get rid of lables like "Invalid" and "wontfix". It's much better to explain why an issue get's closed instead of just marking it and essentially telling people to "shut up, this won't be fixed, ever"

**Comment by EnTeQuAk** [comment]
Also worth a read: http://www.ianbicking.org/blog/2014/03/use-github-issues-to-organize-a-project.html

**Comment by andymckay** [comment]
Currently since there's no way to assign an issue to an outside contributor, we can't tell easily (from a query) if the bug is assigned to a contributor. Could we get "assigned to contributor" or similar label?

**Comment by muffinresearch** [comment]
I'm going to work on getting a master list together and then from there we can work out a plan to sync that list out to all the relevant repos.

**Comment by muffinresearch** [comment]
Here's a first cut of a styleguide for labels based on consolidating and expanding on all the lists @EnTeQuAk put together:

http://jsbin.com/fiboxe/edit?output

**Comment by muffinresearch** [comment]
This is done and the triage label update is posted here mozilla/addons-bots#15 closing since there's nothing else to do. 

Thanks everyone for your input!
